app/controllers/Admin.scala

```diff
@@ -207,5 +212,87 @@ object Admin extends Controller with PanDomainAuthActions {
     )
   }
 
+  /*
+    Section Tag association
+   */
+
+  val addSectionTagForm = Form(
+    mapping(
+      "section_id" -> longNumber,
+      "tag_id" -> nonEmptyText
+    )(assignTagToSectionFormData.apply)(assignTagToSectionFormData.unapply)
+  )
+
+  val removeSectionTagForm = Form(
+    mapping(
+      "section_id" -> longNumber,
+      "tag_id" -> nonEmptyText
+    )(unAssignTagToSectionFormData.apply)(unAssignTagToSectionFormData.unapply)
+  )
+
+  def sectionsAndTags(selectedSectionIdOption: Option[Long]) = (AuthAction andThen WhiteListAuthFilter).async {
+
+    val selectedSectionOptionFt:Future[Option[Section]] = selectedSectionIdOption match {
+      case Some(selectedSectionId) => {
+        for {
+          sections <- getSortedSections()
+        }yield{
+          Some(sections.filter( x => x.id == selectedSectionId ).head)
+        }
+      }
+      case None => Future(None)
+    }
+    val tagIdsOptionFuture: Future[Option[List[String]]] = selectedSectionIdOption match {
```

Comment:
There's not much need to return a Future[Option[List[String]]] here, you could instead return a Future[List[String]] by flatmapping depending on how you use it throughout the rest of your code.